src/app/models/operation.model.ts

```diff
@@ -1,43 +1,106 @@
+import {Node} from "./node.model";
 import {Trainrun} from "./trainrun.model";
 import {TrainrunSection} from "./trainrunsection.model";
 
 enum OperationType {
   create = "create",
   update = "update",
-  delete = "delete"
+  delete = "delete",
 }
 
-export abstract class Operation {
+enum OperationObjectType {
+  trainrun = "trainrun",
+  trainrunSection = "trainrunSection",
+  trainrunSections = "trainrunSections",
+  node = "node",
+  nodes = "nodes",
+};
+
+abstract class Operation<T> {
   readonly type: OperationType;
+  readonly objectType: OperationObjectType;
+
+  readonly object: T;
+  readonly param: any;
 
-  constructor(type: OperationType) {
+  constructor(type: OperationType, objectType: OperationObjectType, object: T) {
     this.type = type;
+    this.objectType = objectType;
+    this.object = object;
+    console.log(this);
   }
 }
 
-export class CreateTrainrunOperation extends Operation {
-  readonly trainrunSection: TrainrunSection;
+class CreateTrainrunOperation extends Operation<Trainrun> {
```

Comment:
We can have a single `TrainrunOperation` class, instead of 3 different ones for create/update/delete.

Reply:
done